llvm/lib/Target/TargetMachine.cpp

```diff
@@ -26,6 +26,11 @@
 #include "llvm/Target/TargetLoweringObjectFile.h"
 using namespace llvm;
 
+cl::opt<bool> KernelInfoEndLTO(
```

Comment:
Can you just unconditionally run the pass, and early exit if !isAnalysisRemarkEnabled?

Reply:
Done.  But I converted -kernel-info-end-lto to -no-kernel-info-end-lto so it's still possible to experiment with positioning kernel-info elsewhere in a pass pipeline, as described in KernelInfo.rst.